### ISSUE TITLE
Trying to plot only error bars (with no line connecting data points) results in empty plot

### ISSUE THREAD

**Issue by anntzer** [matplotlib/matplotlib#8117]
mpl2.0, regression from 1.5.3 and thus marked as release critical, but feel free to relabel/remilestone.
```
plt.errorbar([1, 1.6], [3, 4], [5, 6], capsize=2, fmt="none")
```
1.5.3
![153](https://cloud.githubusercontent.com/assets/1322974/23156933/d49c1ada-f7cd-11e6-85a5-bfec6aacf7e2.png)

2.0
![20](https://cloud.githubusercontent.com/assets/1322974/23156935/d8f17616-f7cd-11e6-8f01-083a2cf60cca.png)

The error bars are not visible in 2.0.  Further inspection indicates that they are, in fact, drawn with the same color as the data line (that connects each data point), i.e., (0, 0, 0, 0).

Note that the docstring of errorbar is a bit inconsistent (emphasis mine):
> fmt : plot format string, optional, default: None
The plot format symbol. **If fmt is 'none' (case-insensitive),
only the errorbars are plotted.  This is used for adding
errorbars to a bar plot, for example.**
>ecolor : mpl color, optional, default: None
A matplotlib color arg which gives the color the errorbar lines;
**if None, use the color of the line connecting the markers.**

A strict reading of the doc for `ecolor` suggests that the current behavior is "correct", but the doc of `fmt` clearly indicates that this is not the intent.

**Comment by tacaswell** [comment]
Agree this should be release critical (and my fault :disappointed: )

**Comment by dstansby** [comment]
`git-bisect` tells me that this appeared in #5593

**Comment by dstansby** [comment]
Just bumped into this doing day-to-day work - a workaround is

`plt.errorbar(x, y, fmt='none, ecolor=ebarcolor)`

where `ebarcolor` is the colour desired for the errorbars.

**Comment by dstansby** [comment]
(re. doc inconsistency, the difference is `None` (not a string) and `'none'` (a string))

**Comment by tacaswell** [comment]
what is the doc inconsistency between `None` and `'none'`?  I think we are pretty consistent about `'none'` -> no style and `None` -> default style.

**Comment by dstansby** [comment]
According to the code, using `None` is deprecated since 1.4 anyway:

```
    if fmt is None:
            fmt = 'none'
            msg = ('Use of None object as fmt keyword argument to ' +
                   'suppress plotting of data values is deprecated ' +
                   'since 1.4; use the string "none" instead.')
            warnings.warn(msg, mplDeprecation, stacklevel=1)

```

**Comment by tacaswell** [comment]
closed by #8163
